### PR TITLE
Briefly document major differences relative to Lex/Yacc.

### PR DESCRIPTION
**Pull request by ltratt**
This is intended to start addressing https://github.com/softdevteam/grmtools/issues/133#issuecomment-589827654 -- we could, and probably should, do more, but this is a start at least.

### CODE REVIEW

File: doc/src/lexcompatibility.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Lex compatibility
+
+grmtools currently supports one common use of Lex, which is to produce a
+sequence of tokens. All Lex files require at least some porting to grmtools,
+though in many cases this is fairly trivial. Nevertheless aspects such as
```
Comment (ptersilie) [inline doc/src/lexcompatibility.md]: `,` after `nevertheless`?

File: doc/src/lexcompatibility.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+There are several major differences between Lex and grmtools:
+
+ * Lex has its own regular expression language whereas grmtools uses the well
+   known Rust [regex crate](https://crates.io/crates/regex) for regular
+   expressions. These two regular expression languages are very similar, but
+   complex regular expressions might not be supported under one or the other.
+
+ * Lex files consist of a sequence of regular expressions and an action for each.
+   grmtools lex files consists of a sequence of regular expressions and simple
```
Comment (ptersilie) [inline doc/src/lexcompatibility.md]: s/simple/simply/

File: doc/src/lexcompatibility.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+   a token name. Actions are not currently supported (and, by extension, nor
+   are special action expressions such as `ECHO` and `REJECT`).
+
+ * Start conditions, character sets, and changes to internal array sizes are
```
Comment (ptersilie) [inline doc/src/lexcompatibility.md]: With `Start conditions`, do you mean context sensitivity?

Comment (ltratt) [reply doc/src/lexcompatibility.md]: It's a Lex feature: I've never thought much about it to be honest, so I don't know it in detail.

**Comment by ltratt** [comment]
Fixed in https://github.com/softdevteam/grmtools/pull/158/commits/239d85c398f724627000a6fa6915c218d90231e2.

File: doc/src/lexcompatibility.md (hunk 3 above)
Comment (ptersilie) [reply doc/src/lexcompatibility.md]: I think we should also mention that we do not support context sensitivity, which Lex does:

> Sometimes it is desirable to have several sets of lexical rules to be applied at different times in the input. For example, a compiler preprocessor might distinguish preprocessor statements and analyze them differently from ordinary statements.

Comment (ptersilie) [reply doc/src/lexcompatibility.md]: Actually, nevermind. "Start conditions" is used for context-sensitivity (see Chapter 10 in the Lex doc).

**Comment by ltratt** [comment]
OK to squash?

**Comment by ptersilie** [comment]
Please squash.

**Comment by ltratt** [comment]
Squashed.

**Pull request merged**